rasa/nlu/regex_message_handler.py

```diff
@@ -0,0 +1,283 @@
+from __future__ import annotations
+import json
+from json.decoder import JSONDecodeError
+import logging
+import re
+from typing import Any, Dict, Match, Optional, Pattern, Text, List
+
+from rasa.engine.graph import GraphComponent, ExecutionContext
+from rasa.engine.storage.resource import Resource
+from rasa.engine.storage.storage import ModelStorage
+from rasa.shared.core.domain import Domain
+
+# FIXME: https://rasa.com/docs/rasa/stories/ doesn't actually explain this format?...
+from rasa.shared.constants import DOCS_URL_STORIES, INTENT_MESSAGE_PREFIX
+from rasa.shared.nlu.constants import (
+    ENTITIES,
+    ENTITY_ATTRIBUTE_TYPE,
+    ENTITY_ATTRIBUTE_VALUE,
+    INTENT,
+    INTENT_NAME_KEY,
+    INTENT_RANKING_KEY,
+    PREDICTED_CONFIDENCE_KEY,
+    TEXT,
+)
+import rasa.shared.utils.io
+from rasa.shared.nlu.training_data.message import Message
+
+logger = logging.getLogger(__name__)
+
+
+class RegexMessageHandler(GraphComponent):
```

Comment:
should we put this into the `nlu.classifier` package as it's kind of like the `DIETClassifier` ?

Reply:
good idea - Btw, I just had a look at the entity featurization (for #9205), and I think no *ML* policy does anything with the entities extracted here 😅   (since there are no tokens - which means all the "entities to index" functions that are included in `SingleStateFeaturizer` won't do anything with these entities). Rule/Memoization should be able to handle them 🤔  ....  Anyway, will move to classifier! 😄

Reply:
lol 🙈 cc @JEM-Mosig As we have considered removing the entity featurization with the global slot mappings

Reply:
MemoizationPolicy can definitely distinguish by entity. RulePolicy doesn't.